File: src/main/java/me/devsnox/battlekits/kits/kit/KitType.java
package me.devsnox.battlekits.kits.kit;

import org.bukkit.ChatColor;

public enum KitType {

    RARE(ChatColor.DARK_AQUA, 3, "Selten", 30.0D),
    EPIC(ChatColor.DARK_PURPLE, 2, "Episch", 25.0D),
    LEGENDARY(ChatColor.GOLD, 1, "Legendär", 20.0D),
    MYTHIC(ChatColor.RED, 14,"Mystisch", 15.0D);

    private ChatColor color;
    private int durability;
    private String name;
    private double chance;

    KitType(ChatColor color, int durability, String name, double chance) {
        this.color = color;
        this.durability = durability;
        this.name = name;
        this.chance = chance;
    }

    public ChatColor getColor() {
        return color;
    }

    public int getDurability() {
        return this.durability;
    }

    public String getName() {
        return name;
    }

    public double getChance() {
        return chance;
    }
}

File: spigot/src/main/kotlin/me/devsnox/battlekits/kits/kit/KitType.kt
package me.devsnox.battlekits.kits.kit

import de.tr7zw.itemnbtapi.NBTItem
import org.bukkit.ChatColor

enum class KitType(
    val color: ChatColor,
    val displayName: String,
    val chance: Double
) {
    RARE(ChatColor.DARK_AQUA, "Selten", 30.0),
    EPIC(ChatColor.DARK_PURPLE, "Episch", 25.0),
    LEGENDARY(ChatColor.GOLD, "Legendär", 20.0),
    MYTHIC(ChatColor.RED, "Mystisch", 15.0)
}

fun NBTItem.toKitType(): KitType? = KitType.values().find { it.displayName == getString("type") }
